client/driver/docker.go

```diff
@@ -118,6 +118,22 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 	logger.Printf("[DEBUG] driver.docker: using %d bytes memory for %s", hostConfig.Memory, task.Config["image"])
 	logger.Printf("[DEBUG] driver.docker: using %d cpu shares for %s", hostConfig.CPUShares, task.Config["image"])
 
+	mode := task.Config["network_mode"]
```

Comment:
You can check if the value exists in the map using the tupled return type like: `if mode, ok := task.Config["network_mode"]`. Should make using the default cleaner.